### ISSUE TITLE
How to not get errors due to ejs templates

### ISSUE THREAD

**Issue by markfaine** [htmllint/htmllint#220]
Is there a way to not validate and not produce errors caused by the use of EJS templates?  

For example:

```
  <title>
    <%= htmlWebpackPlugin.options.title %>
  </title>
```

**Comment by mlochbaum** [comment]
Setting `raw-ignore-regex` (introduced version 0.7.0) to `\<%=[^]*?\>` will remove everything that looks like `<%=...%>` before parsing or linting. Does that work for you?

**Comment by markfaine** [comment]
No, I've already tried multiple variations of that with no success.

Here is my config file:
{
    "plugins": [],
    "attr-name-style": false,
    "indent-style": "spaces",
    "indent-width": "2",
    "line-end-style": "lf",
    "label-req-for": true,
    "input-req-label": true,
    "id-no-dup": true,
    "html-req-lang": true,
    "doctype-first": true,
    "attr-no-unsafe-char": true,
    "attr-no-dup": true,
    "raw-ignore-regex": /\<%=[^]*?\>/
}

Here is the error:
/home/mark/node-v8.7.0-linux-x64/lib/node_modules/htmllint-cli/node_modules/cjson/index.js:236
        throw err;
        ^

SyntaxError: No value found for key raw-ignore-regex at 14:25
    "raw-ignore-regex": /\<%=[^]*?\>/
                        ^
File: "/home/mark/Personal/Projects/lint-js-starter/.htmllintrc"
    at Object.parseJSON (/home/mark/node-v8.7.0-linux-x64/lib/node_modules/htmllint-cli/node_modules/jju/lib/parse.js:745:13)

Thank you,
-Mark

**Comment by mlochbaum** [comment]
JSON doesn't support RegExp values. Use the string `"\<%=[^]*?\>"` instead.

**Comment by dennisbaskin** [comment]
JSON will not allow escaping either, but since this is a string, you can just leave the escapes out entirely. The following will capture all tag types: `['<%', '<%_', '<%=', '<%-', '<%#', '<%%']`

```
{
  "raw-ignore-regex": "<%[^]*?%>|<%%"
}
```

And if you have multiples (e.g. on same line), it won't matter. Internally it uses a global flag for the regex:

```
    regexGlobal: function (r) {
        r = module.exports.regex(r);
        return r && new RegExp(r.source, r.ignoreCase ? 'gi' : 'g');
    },
```

A simple example of this (you can test in browser console):

```
const regex = RegExp("<%[^]*?%>|<%%", 'g')
const string = `
  <% if (controlFlow) { %>
    <div>
      <%- include('./partial') %>
    </div>

    <strong><%= some.value %>" and <%_ some.otherValue %></strong>
    <%# some random comment %> and
  <% } %>
  
  escape for literal: <%% 
`

string.match(regex)
```

The tags are directly from [ejs docs](http://ejs.co/#docs) under the "Tags" section
